dlls/winex11.drv/mouse.c

```diff
@@ -551,8 +551,10 @@ BOOL clip_fullscreen_window( HWND hwnd, BOOL reset )
     release_win_data( data );
     if (!fullscreen) return FALSE;
     if (!(thread_data = x11drv_thread_data())) return FALSE;
-    if (GetTickCount() - thread_data->clip_reset < 1000) return FALSE;
-    if (!reset && clipping_cursor && thread_data->clip_hwnd) return FALSE;  /* already clipping */
+    if (!reset) {
+        if (GetTickCount() - thread_data->clip_reset < 1000) return FALSE;
```

Comment:
I do not understand why clip_reset is needed here. It probably should be just removed.

Reply:
This change makes sense to me and could maybe go upstream.

Reply:
The only thing I don't like about it is that I make the change specifically to fix cases when the resolution is changed, i.e. when clip_fullscreen_window() is called from X11DRV_resize_desktop(). But reset is also true when clip_fullscreen_window() is called X11DRV_ClipCursor(). It should not matter with the vanilla wine because it never called if the mouse is not already clipped, but it would matter if b0ad45c47a167fb149f9f12bb1b24e0edaaaeb62 is applied.

But like I said, I don't understand why clip_reset is here, so I cannot judge when we can safely ignore it.